packages/wrapped-keys-lit-actions/src/lib/ethereum/internal/signMessage.js

```diff
@@ -0,0 +1,39 @@
+/* global ethers */
+
+async function signMessage({ privateKey, messageToSign }) {
+  try {
+    const wallet = new ethers.Wallet(privateKey);
+    const signature = await wallet.signMessage(messageToSign);
+
+    return { signature, walletAddress: wallet.address };
+  } catch (err) {
+    throw new Error(`When signing message - ${err.message}`);
+  }
+}
+
+function verifyMessageSignature({ messageToSign, signature }) {
+  try {
+    return ethers.utils.verifyMessage(messageToSign, signature);
+  } catch (err) {
+    throw new Error(
+      `When validating signed Ethereum message is valid: ${err.message}`
+    );
+  }
+}
+
+export async function signMessageEthereumKey({ privateKey, messageToSign }) {
+  const { signature, walletAddress } = await signMessage({
+    privateKey,
+    messageToSign,
+  });
+
+  const recoveredAddress = verifyMessageSignature({ messageToSign, signature });
```

Comment:
**LATER:** Ideally this shouldn't be required and this adds an additional overhead as well, not sure by how much though. So it'll be great if we can add timings to see this

Reply:
I'm happy to remove it if it's superfluous -- I only copied it in because it was in the existing implementation :+1:  Is there any case where this would fail? Usually verification of the signature is the responsibility of the receiver of the signature rather than the creator of it and it does appear that this should always work unless ethers is broken internally for some reason (at which point the consumer's verify should fail anyway?)